packages/cw-orch-mock/src/core.rs

```diff
@@ -145,15 +174,24 @@ impl Mock<MockState> {
 impl<S: StateInterface> Mock<S> {
     /// Create a mock environment with a custom mock state.
     /// The state is customizable by implementing the `StateInterface` trait on a custom struct and providing it on the custom constructor.
-    pub fn new_custom(sender: &Addr, custom_state: S) -> Self {
+    pub fn new_custom(sender: impl Into<String>, prefix: &'static str, custom_state: S) -> Self {
         let state = Rc::new(RefCell::new(custom_state));
-        let app = Rc::new(RefCell::new(custom_app::<Empty, Empty, _>(|_, _, _| {})));
+        let app = Rc::new(RefCell::new(
+            AppBuilder::new_custom()
+                .with_api(MockApiBech32::new(prefix))
+                .with_wasm(WasmKeeper::new().with_address_generator(MockAddressGenerator))
+                .build(|_, _, _| {}),
+        ));
+        let sender = app.borrow().api().addr_make(&sender.into()).clone();
```

Comment:
Let's add log of sender address here? @CyberHoward Do you think it would be enough if we cover generated addresses with logs?

Reply:
I think a log will be a bit useless as it's logged on creation of the `Mock`. Ideally you want to have the human readable representation of the address whenever the call is made, which won't be easy with this.

I suggest we just leave it as it is and see how annoying it gets.

Reply:
However, I do think that this line should be changed. 

If sender is already a valide bech32 we should not re-hash it. 

For example, in our migration tests (clone testing) we set up a mock environment and use the actual wallet address as the sender. With this impl that wallet address will be re-hashed and the resulting address won't exist on-chain.

Reply:
So we should use the api's `addr_validate` to check if it's valid and only re-hash the invalid addresses.

Reply:
That's a good idea. However, is that predictable for users ? 
For clone testing, we're not using this code anyway, so no issues there :)

Reply:
I think I'd be very confused if I gave it a valid bech32 and it turned out to become a different address.

Reply:
I get why they needed to make this change for instantiate2 but for testing it **sucks**

Reply:
Yes.
I'm not sure we **have** to make this change though. Investigating